File: common/src/main/java/org/sandboxpowered/silica/network/clientbound/PongResponse.java
package org.sandboxpowered.silica.network.clientbound;

import org.sandboxpowered.silica.network.Packet;
import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;

public class PongResponse implements Packet {
    private long time;

    public PongResponse(long time) {
        this.time = time;
    }

    public PongResponse() {
    }

    @Override
    public void read(PacketByteBuf buf) {
        this.time = buf.readLong();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeLong(this.time);
    }

    @Override
    public void handle(PacketHandler packetHandler) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/vanilla/network/handshake/serverbound/PongResponse.kt
package org.sandboxpowered.silica.vanilla.network.handshake.serverbound

import org.sandboxpowered.silica.vanilla.network.Connection
import org.sandboxpowered.silica.vanilla.network.Packet
import org.sandboxpowered.silica.vanilla.network.PacketByteBuf
import org.sandboxpowered.silica.vanilla.network.PacketHandler

class PongResponse(private val time: Long) : Packet {
    constructor(buf: PacketByteBuf) : this(buf.readLong())

    override fun write(buf: PacketByteBuf) {
        buf.writeLong(time)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {}
}